odl/discr/lp_discr.py

```diff
@@ -221,18 +229,88 @@ def element(self, inp=None):
                         ''.format(arr.shape, self.shape))
             arr = arr.ravel(order=self.order)
             return self.element_type(self, self.dspace.element(arr))
-        except TypeError as err:
+        except (TypeError, IndexError) as err:
             if str(err.args[0]).startswith('output contains points outside'):
```

Comment:
Second thought, the error raised in `fspace` is a `ValueError` anyway and will propagate through, so I'll just delete this part. Alternatively, I could add `ValueError` and append the error message just as in the other two cases.